packages/react-next/src/components/Modal/Modal.base.tsx

```diff
@@ -183,12 +90,30 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
       onLayerDidMount,
       isModeless,
       dragOptions,
-    } = this.props;
-    const { isOpen, isVisible, hasBeenOpened, modalRectangleTop, x, y, isInKeyboardMoveMode } = this.state;
+      onDismissed,
+    } = props;
 
-    if (!isOpen) {
-      return null;
-    }
+    const internalState = useConst<IModalInternalState>(() => ({
+      onModalCloseTimer: 0,
+      allowTouchBodyScroll: false,
+      hasRegisteredKeyUp: false,
+      scrollableContent: null,
+      lastSetXCoordinate: 0,
+      lastSetYCoordinate: 0,
+      events: new EventGroup({}),
+    }));
+
+    const [isModalOpen, { setFalse: setModalClose, setTrue: setModalOpen }] = useBoolean(isOpen);
+
+    const [isModalMenuOpen, { toggle: toggleModalMenuOpen, setFalse: setModalMenuClose }] = useBoolean(false);
+
+    const [hasBeenOpened, { setTrue: setHasOpenedTrue }] = useBoolean(isOpen);
```

Comment:
It's passed into getClassNames. When Modal is opened for the first time and the topOffsetFixed prop is true it will set root's alignItems to 'flex-start' and set main's top to modalRectangleTop.